tgui/packages/tgui/interfaces/LuaEditor/index.js

```diff
@@ -9,180 +9,217 @@ import { Log } from './Log';
+
+    window.addEventListener('resize', () => this.setState({}));
+  }
+
+  render() {
+    const { act, data } = useBackend(this.context);
+    const { noStateYet, globals, documentation, tasks } = data;
+    const [modal, setModal] = useLocalState(
+      this.context,
+      'modal',
+      noStateYet ? 'states' : null
+    );
+    const [activeTab, setActiveTab] = useLocalState(
+      this.context,
+      'activeTab',
+      'globals'
+    );
+    const [input, setInput] = useLocalState(this.context, 'scriptInput', '');
+    let tabContent;
+    switch (activeTab) {
+      case 'globals': {
+        if (!globals) {
+          tabContent = (
+            <h1>
+              Could not retrieve the global table. Was it corrupted or shadowed?
+            </h1>
+          );
+        } else {
+          tabContent = (
+            <ListMapper
+              list={globals}
+              skipNulls
+              vvAct={(path) => act('vvGlobal', { indices: path })}
+              callType="callFunction"
+            />
+          );
+        }
+        break;
       }
-      break;
-    }
-    case 'tasks': {
-      if (!tasks) {
-        tabContent = (
-          <h1>
-            Could not retrieve task info. Was the global table corrupted or
-            shadowed?
-          </h1>
-        );
-      } else {
-        tabContent = <TaskManager />;
+      case 'tasks': {
+        if (!tasks) {
+          tabContent = (
+            <h1>
+              Could not retrieve task info. Was the global table corrupted or
+              shadowed?
+            </h1>
+          );
+        } else {
+          tabContent = <TaskManager />;
+        }
+        break;
+      }
+      case 'log': {
+        tabContent = <Log />;
+        break;
       }
-      break;
-    }
-    case 'log': {
-      tabContent = <Log />;
-      break;
     }
-  }
-  return (
-    <Window width={1280} height={720}>
-      <Window.Content>
-        <Button icon="file" onClick={() => setModal('states')}>
-          States
-        </Button>
-        {noStateYet ? (
-          <Flex
-            width="100%"
-            height="100%"
-            align="center"
-            justify="space-around">
-            <h1>Please select or create a lua state to get started.</h1>
-          </Flex>
-        ) : (
-          <Stack>
-            <Stack.Item>
-              <Section
-                fill
-                title="Input"
-                buttons={
-                  <>
-                    <Button.File
-                      onSelectFiles={(file) => setInput(file)}
-                      accept=".lua,.luau">
-                      Import
-                    </Button.File>
-                    <Button onClick={() => setModal('documentation')}>
-                      Help
+    const jumpButtonCondition =
+      activeTab === 'log' &&
+      this.sectionRef.current?.scrollableRef.current.scrollHeight >
+        this.sectionRef.current?.scrollableRef.current.clientHeight;
```

Comment:
you should hook onto onScroll instead and make this a state, otherwise it'll fail to update if you scroll up